File: backend/services/core/src/main/java/app/mnema/core/media/config/MediaClientConfig.java
package app.mnema.core.media.config;

import org.springframework.boot.context.properties.EnableConfigurationProperties;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.client.RestClient;

@Configuration
@EnableConfigurationProperties(MediaClientProps.class)
public class MediaClientConfig {

    @Bean
    public RestClient mediaRestClient(MediaClientProps props) {
        return RestClient.builder()
                .baseUrl(props.baseUrl())
                .build();
    }
}

File: backend/services/user/src/main/kotlin/app/mnema/user/media/config/MediaClientConfig.kt
package app.mnema.user.media.config

import org.springframework.boot.context.properties.EnableConfigurationProperties
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.web.client.RestClient

@Configuration
@EnableConfigurationProperties(MediaClientProps::class)
class MediaClientConfig {
    @Bean
    fun mediaRestClient(props: MediaClientProps): RestClient =
        RestClient.builder()
            .baseUrl(props.baseUrl)
            .build()
}
